src/AMSlib/wf/workflow.hpp

```diff
@@ -121,7 +121,11 @@ class AMSWorkflow
     bool *hPredicate = nullptr;
 
     if (appDataLoc == AMSResourceType::AMS_HOST) {
+#ifdef __ENABLE_MPI__
+      return DB->store(num_elements, inputs, outputs, comm, predicate);
```

Comment:
It is unclear to me why comm is required when storing data. I will look closer in the code. It feels though that we start adding a secondary "meaning" to the store command.

Reply:
Yes it is not the cleaner implementation, it leads to a lot of conditional code which I don't really like. I chose to pass down the MPI comm to encode the actual MPI rank in AMSMessage. I considered an alternative implementation where I would generate an random ID for each rank use it in AMSMessage, the advantage is a cleaner implementation because `store` would not need any MPI_Comm. We could also choose to not encode MPI rank in AMSMessage.

Let me know your thoughts about it.

Reply:
Wouldn't it be enough to pass the rank-id when constructing the DB object itself?

Reply:
[Here](https://github.com/LLNL/AMS/blob/features/mlmodel-update/src/AMSlib/wf/workflow.hpp#L210) you have the rId. Over [here](https://github.com/LLNL/AMS/blob/features/mlmodel-update/src/AMSlib/wf/workflow.hpp#L227)  we pass it to the DB manager. I can see also [here](https://github.com/LLNL/AMS/blob/features/mlmodel-update/src/AMSlib/wf/basedb.hpp#L1996) that we forward the rank id to RMQ.

Reply:
You are right I missed the obvious solution... I refactored everything and removed `comm` from `store` methods!